smartctl_lld/__init__.py

```diff
@@ -0,0 +1,403 @@
+    else:
+        sender.append('%s smartctl.info[%s,SmartStatus] "NO_SMART_VALUES"' % (host, dR))   # something else
+
+    return sender
+
+
+def getAllDisks(config, host, command, diskList):
+    driveHeaders = []
+    jsonData = []
+    senderData = []
+    for d in diskList:   # cycle through disks
+        getSmart_Out = getSmart(config, host, command, d)
+        finalD = getSmart_Out[4][0]
+        origD = getSmart_Out[4][1]
+
+        # Check for smartctl binary-related errors (if disk scan was bypassed)
+        if getSmart_Out[0]:
+            configError = getSmart_Out[0]   # D_OS_NOCMD, D_OS_ERROR; may rewrite previous (similar) error
+            break   # fatal error
+
+        # Begin duplicate check if desired
+        if config['skipDuplicates']:
+            serialCurrent = getSmart_Out[3][0]
+            headerCurrent = getSmart_Out[3][1]
+
+            if serialCurrent and headerCurrent:   # if serial and secondary identifying data is found
+                duplicate = False
+                for s,h in driveHeaders:
+                    if s == serialCurrent and h == headerCurrent:
+                        senderData.append('%s smartctl.info[%s,DriveStatus] "DUPLICATE"' % (host, origD))
+                        jsonData.append({'{#DDRIVESTATUS}':origD})   # populate duplicate only with original drive ID
+                        duplicate = True
+                        break   # break from header check
+
+                if duplicate == True:
+                    continue   # continue to the next disk. other sender and json data is discarded
+
+                driveHeaders.append((serialCurrent, headerCurrent))   # add header info for current disk, only if it's not duplicate
+
+        # Add collected data
+        jsonData.append({'{#DDRIVESTATUS}':finalD})   # always populate 'DriveStatus' LLD
+        senderData.extend(getSmart_Out[1])
+        jsonData.extend(getSmart_Out[2])
+    return jsonData, senderData
+
+# Read the config file and return a dict of settings
+def parseConfig(path=None):
+    if sys.platform == 'linux':
+        agent_conf = '/etc/zabbix/zabbix_agentd.conf'
+        sender_path = "zabbix_sender"
+        sender_py_path = "/etc/zabbix/scripts/sender_wrapper.py"
+        if not path:
+            path = "/etc/zabbix/zabbix-smartmontools.conf"
+    elif 'freebsd' in sys.platform:
+        # FreeBSD usually installs Zabbix in a dir containing the version
+        # number, ala "zabbix42"
+        try:
+            zabbixdir = glob.glob('/usr/local/etc/zabbix*')[0]
+        except IndexError:
+            raise 'Error: can\'t find path to Zabbix config directory'
+        agent_conf = '%s/zabbix_agentd.conf' % zabbixdir
+        sender_path = "zabbix_sender"
+        sender_py_path = "%s/scripts/sender_wrapper.py" % zabbixdir
+        if not path:
+            path = "%s/zabbix-smartmontools.conf" % zabbixdir
+    elif os.name == 'nt':
```

Comment:
```suggestion
    elif sys.platform == 'win32':
```
Essential for Windows, don't remember why though.

Reply:
What value does `sys.platform` have on 64-bit windows?  Is it still `win32`?

Reply:
Still `win32`.